src/frontend/web_application/src/components/MultidimensionalPi/components/Ratings/index.jsx

```diff
@@ -2,36 +2,41 @@ import React, { PropTypes } from 'react';
 import classnames from 'classnames';
 import './style.scss';
 
-const Rating = ({ name, level, piMax, className }) => {
+function classFor(element, mini) {
+  const classNames = [`m-pi-ratings__${element}`];
+  if (mini) { classNames.push(`m-pi-ratings--mini__${element}`); }
+
+  return classNames;
+}
+
+const Rating = ({ name, level, piMax, className, mini }) => {
   const width = (level / piMax) * 100;
   const style = { width: `${width}%` };
-  const ratingClassName = classnames(
-    'm-pi-ratings__item',
-    className,
-  );
 
   return (
-    <div className={ratingClassName}>
-      <div className="m-pi-ratings__item-name">
-        <span className="m-pi-ratings__item-name-label">{name}</span>
+    <div className={classnames(classFor('item', mini), className)}>
```

Comment:
I found the declarative way a bit confusing with all those looong classNames that's why I did a func for it. But I agree with your argument, this is not good practice for development.